src/ESports.ts

```diff
@@ -0,0 +1,172 @@
+        }
+
+        const tomorrow = new Date();
+        tomorrow.setHours(tomorrow.getHours() + 12);
+        tomorrow.setMinutes(0);
+
+        let tellDate = "";
+
+        // filter to only show new games (up-to one day in advance)
+        const prints: Map<string, ESportsLeagueSchedule[]> = new Map();
+        for (const [dateKey, entries] of this.schedule.entries()) {
+            if (new Date(dateKey) > tomorrow) break;
+
+            tellDate = dateKey;
+            for (const [league, entryList] of entries) {
+                for (const item of entryList) {
+
+                    if (!item.time.includes("ago")) {
+                        if (!prints.get(league)) {
+                            prints.set(league, Array());
+                        }
+
+                        prints.get(league)!.push(item);
+                    }
+                }
+            }
+        }
+
+        this.sendPrintout(esports as Discord.TextChannel, prints, tellDate);
+        setTimeout(this.postInfo, 1000 * 60 * 60 * 1); // once per hour
+    }
+
+    private sendPrintout(channel: Discord.TextChannel, data: Map<string, ESportsLeagueSchedule[]> | undefined, date: string) {
+
+        if (!data) {
+            channel.send("No games played on this date.");
+            return;
+        }
+
+        let output = `Games being played ${date.split(" ").join("/")}:\n\`\`\``;
+        const padLeague = Array.from(data!.keys()).reduce((a, b) => a.length > b.length ? a : b).length;
+
+        for (const [league, games] of data!) {
+            for (const game of games) {
+                output += `[${league.padEnd(padLeague)}] ${game.teamA.padEnd(3)} vs ${game.teamB.padEnd(3)} -- ${game.time}\n`;
+            }
+        }
+        output += "```";
+
+        channel.send(output);
+    }
+
+    // this can also check for older games by using resultsHtml instead of fixures
+    private async loadData() {
+        // pull data
+        const data = await fetch("https://eu.lolesports.com/en/api/widget/schedule?timezone=Europe%2FOslo&leagues=26&leagues=3&leagues=2&leagues=6&leagues=7&leagues=5&leagues=4&leagues=9&leagues=10&leagues=1&leagues=43&slug=all");
+        const html = (await data.json() as ESportsAPIReturnData).fixturesHtml;
+        const schedule: Map<string, Map<string, ESportsLeagueSchedule[]>> = new Map();
+
+        // for each date
+        const asHtml = CheerioAPI.load(html);
+        const parents = asHtml.root().find(".schedule__row-group");
+        parents.each((indexed, elem) => {
+            // the current date
+            const elemRoot = CheerioAPI.load(elem).root();
+            const date = elemRoot.find(".schedule__row--date").find("h2").text().split(" ");
+
+            // hack to get month number from text
+            const month = ("0" + ("JanFebMarAprMayJunJulAugSepOctNovDec".indexOf(date[2].substr(0, 3)) / 3 + 1)).slice(-2);
+
+            const realDate = `${new Date().getFullYear()} ${month} ${date[1]}`;
```

Comment:
This might be incorrect if games are played in january while we're in december.

Reply:
Not sure how that case should be handled 🤔